powerauth-restful-security-spring/src/main/java/io/getlime/security/powerauth/rest/api/spring/service/v3/ActivationService.java

```diff
@@ -339,4 +323,27 @@ private void handleInvalidRecoveryError(SoapFaultDetail faultDetail) throws Powe
             throw new PowerAuthRecoveryException(errorMessage, "INVALID_RECOVERY_CODE", currentRecoveryPukIndex);
         }
     }
+
+
+    /**
+     * Prepare payload for the encrypted response.
+     *
+     * @param encryptedData Encrypted data.
+     * @param mac MAC code of the encrypted data.
+     * @param processedCustomAttributes Custom attributes to be returned.
+     * @return Encrypted response object.
+     */
+    private ActivationLayer1Response prepareEncryptedResponse(String encryptedData, String mac, Map<String, Object> processedCustomAttributes) {
```

Comment:
This whole method is a duplicate of `ActivationService.prepareEncryptedResponse` in `powerauth-restful-security-javaee`, it could be fixed by common utils in `powerauth-restful-security-base` but it is rather nitpick for now.

Reply:
I will now merge this as is - it seems to me that the additional inheritance (like `AbstractActivationService` with just an abstract method and this one protected method) or a separate "utils" class would make the code more complicated than this simple private utility method.